splink/linker.py

```diff
@@ -1172,10 +1164,14 @@ def predict(
 
         # _initialise_df_concat_with_tf returns None if the table doesn't exist
         # and only SQL is queued in this step.
-        input_dataframes = self._initialise_df_concat_with_tf(
-            materialise=False, return_as_list=True
+        nodes_with_tf = self._initialise_df_concat_with_tf(
+            materialise=materialise_after_computing_term_frequencies
         )
 
+        input_dataframes = []
+        if nodes_with_tf:
+            input_dataframes.append(nodes_with_tf)
```

Comment:
So, the idea here is that we keep a running list of input_dataframes, instead of just passing `nodes_with_tf` in?

Reply:
`nodes_with_tf` could be null, so we need to check this before adding it to the list of data frames that are sent into execute SQL pipeline.  I got rid of the None handling logic from the pipeline because I think it's confusing to be able to pass either Splinkdataframes or Nones - if nothing else it makes the type hint harder to write!